modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -94,7 +95,8 @@ CV__DNN_INLINE_NS_BEGIN
         DNN_TARGET_FPGA,  //!< FPGA device with CPU fallbacks using Inference Engine's Heterogeneous plugin.
         DNN_TARGET_CUDA,
         DNN_TARGET_CUDA_FP16,
-        DNN_TARGET_HDDL
+        DNN_TARGET_NPU,
+        DNN_TARGET_HDDL,
```

Comment:
Hi, @alalek. Thank you for code review. In my test, if I move `DNN_BACKEND_TIMVX` to the end of `enum Backend`, I will get an error. 

If I do the following, the value of `DNN_BACKEND_TIMVX` in the OpenCV library is `1000001`, while the value of `DNN_BACKEND_TIMVX` when calling the OpenCV library is `6`.
This may be caused by `#ifdef __OPENCV_BUILD` flag. In this case, I can not use `DNN_BACKEND_TIMVX` to trigger TimVx Backend, I can only use `1000001` to do it.
It happens on `Ubuntu of ARM`, but not for the `Ubuntu of X86`. 

```C++
    enum Backend
    {
        //! DNN_BACKEND_DEFAULT equals to DNN_BACKEND_INFERENCE_ENGINE if
        //! OpenCV is built with Intel's Inference Engine library or
        //! DNN_BACKEND_OPENCV otherwise.
        DNN_BACKEND_DEFAULT = 0,
        DNN_BACKEND_HALIDE,
        DNN_BACKEND_INFERENCE_ENGINE,            //!< Intel's Inference Engine computational backend
                                                 //!< @sa setInferenceEngineBackendType
        DNN_BACKEND_OPENCV,
        DNN_BACKEND_VKCOM,
        DNN_BACKEND_CUDA,
#ifdef __OPENCV_BUILD
        DNN_BACKEND_INFERENCE_ENGINE_NGRAPH = 1000000,     // internal - use DNN_BACKEND_INFERENCE_ENGINE + setInferenceEngineBackendType()
        DNN_BACKEND_INFERENCE_ENGINE_NN_BUILDER_2019,      // internal - use DNN_BACKEND_INFERENCE_ENGINE + setInferenceEngineBackendType()
#endif
        DNN_BACKEND_TIMVX
    };
```

Reply:
This comment is about TARGET, not BACKEND.

Requirement of addition to the end comes from preserving values of already exited enum items.

Reply:
In BACKEND put to the end before `#ifdef __OPENCV_BUILD`

Reply:
Thank your reply, got it.